Sources/Client/Views/Config/TroubleShootingView.swift

```diff
@@ -0,0 +1,205 @@
+          displayBanner(with: "Confirm perform fresh install?") {
+            performFreshInstall { error in
+              DeltaClientApp.fatal("Failed to perform fresh install: \(error)")
+            }
+          }
+        }.buttonStyle(SecondaryButtonStyle())
+      }
+      .frame(width: 200)
+      
+      switch popupState.current {
+      case .shown(let popupObject):
+        PopupView(title: popupObject.title,
+                  subtitle: popupObject.subtitle,
+                  icon: popupObject.image,
+                  action: popupObject.action)
+          .transition(.move(edge: .top))
+          .animation(.easeInOut(duration: popupAnimationDuration))
+      case .hidden: EmptyView()
+      }
+    }
+  }
+  
+  
+  // MARK: - Methods.UI
+  
+  
+  /// Displays a confirmation banner meant for the user to confirm his choice or discard it
+  ///
+  /// If the user confirms his choice, the `onConfirm` callback will be triggered, otherwise the banner will simply be dismissed
+  /// - Postcondition: `popupState` updated to `.shown`
+  /// - Parameters:
+  ///   - title: the banner title
+  ///   - onConfirm: callback triggered if the user confirms his choice
+  private func displayBanner(with title: String, onConfirm: @escaping (() -> Void)) {
+    func updateToShown() {
+      popupState.update(to: .shown(PopupObject(title: "Warning",
+                                               subtitle: title,
+                                               image: Image(systemName: "exclamationmark.triangle"),
+                                               action: (confirm: {
+                                                          popupState.update(to: .hidden)
+                                                          onConfirm()
+                                                        },
+                                                        cancel: { popupState.update(to: .hidden) }
+                                                       )
+                                              )
+                                  ))
+    }
+    
+    switch popupState.current {
+    case .hidden:
+      updateToShown()
+    case .shown(_):
+      // Another alert is currently being displayed. Dismissing it first before displaying the new one
+      popupState.update(to: .hidden)
+      DispatchQueue.main.asyncAfter(deadline: .now() + popupAnimationDuration*1.1) {
+        updateToShown()
+      }
+    }
+  }
+  
+  /// Displays banner which dismisses by itself
+  ///
+  /// - Parameters:
+  ///   - seconds: number of seconds after which the banner should disappear
+  ///   - title: banner title
+  ///   - subtitle: banner subtitle
+  public func displayBannerWithAutodismiss(after seconds: Double = 3,
+                                           title: String,
+                                           subtitle: String,
+                                           image: Image? = Image(systemName: "checkmark.seal")) {
```

Comment:
This is the way I have chosen to do multiline function signatures in Delta Client

```suggestion
  public func displayBannerWithAutodismiss(
    after seconds: Double = 3,
    title: String,
    subtitle: String,
    image: Image? = Image(systemName: "checkmark.seal")
  ) {
```